test/town_legacy.cpp

```diff
@@ -10,6 +10,12 @@
 #include "scenario/town.hpp"
 #include "scenario/scenario.hpp"
 #include "oldstructs.hpp"
+#include "fileio/fileio.hpp"
+#include <vector>
```

Comment:
This doesn't belong in the middle of a bunch of other includes. As per the style guide, system headers should go above all other headers _except_ for the counterpart header, which in this case doesn't exist. So, in other words, this include should go right at the top.